app/uk/gov/hmrc/thirdpartyapplication/domain/models/Event.scala

```diff
@@ -17,24 +17,63 @@
 package uk.gov.hmrc.thirdpartyapplication.domain.models
 
 import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.util.UUID
+import play.api.libs.json._
+import uk.gov.hmrc.play.json.Union
+import uk.gov.hmrc.thirdpartyapplication.models.EventType
 
-trait UpdateApplicationEvent {
+sealed trait UpdateApplicationEvent {
+  def id: UpdateApplicationEvent.Id
   def applicationId: ApplicationId
-  def timestamp: LocalDateTime
-  def instigator: UserId
+  def eventDateTime: LocalDateTime
+  def actor: UpdateApplicationEvent.Actor
 }
 
-trait UpdateApplicationRepositoryEvent extends UpdateApplicationEvent {
-}
-
-trait UpdateApplicationNotificationEvent extends UpdateApplicationEvent {
-  def requester: String
+trait TriggersNotification {
+  self: UpdateApplicationEvent =>
 }
 
 object UpdateApplicationEvent {
+  sealed trait Actor
+
+  case class GatekeeperUserActor(user: String) extends Actor
+  //case class CollaboratorActor(email: String) extends Actor
+  //case class ScheduledJobActor(jobId: String) extends Actor
+  //case class UnknownActor() extends Actor
```

Comment:
Are these for future use?

Reply:
Yes, they will be needed as we add other types of update command